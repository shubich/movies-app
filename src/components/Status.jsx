import React from 'react';

export const Status = () => (
    <div id='search-results'>
        <div className='count'>7 movies found</div>
        <div className='sort'>
            <span>Sort by</span>
            <a href="#" className='link link-default'>release date</a>
            <a href="#" className='link link-primary'>rating</a>
        </div>
    </div>
);