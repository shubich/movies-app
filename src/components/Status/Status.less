@import "../../styles/palette";
@import "../../styles/spaces";
@import "../../styles/media";

.Status {
    background: #CDCDCD;
    padding: @desktop-space;
    input {
        display: none;
    } 
    @media @phone {
        padding: @phone-space;
    }   
}

.Status-Container {
    display: flex;
    flex-flow: row nowrap;
    @media @phone {
        flex-flow: column wrap;
    }
}

.Status-Count {
    flex: 1;
}

.Status-SortBy {
    font-weight: bold;
}

.Status-Link {
    margin-left: 10px;
    cursor: pointer;
    
    text-decoration: none;
    color: black;

    &:hover {
        color: @primary-color;
    }
}

.Status-Link_primary {
    color: @primary-color;
    cursor: default;
}