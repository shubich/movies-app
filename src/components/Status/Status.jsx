import React from 'react';
import PropTypes from 'prop-types';
import LoadingBars from '../LoadingBars';

import './Status.less';

const Status = ({
  fetching,
  match,
  count,
  searchType,
}) => {
  const showResults = count && match.url !== '/' && !fetching;

  return (
    <div className={`Status ${fetching ? 'Status_padding_horizontal' : ''}`}>
      <div className="Container Status-Container">
        {fetching && <LoadingBars color="green" />}
        {
          showResults ?
            <div className="Status-Count">
              {count} {searchType === 'movies' ? 'movies' : 'people'} found
            </div> : null
        }
      </div>
    </div>
  );
};

Status.propTypes = {
  fetching: PropTypes.bool.isRequired,
  count: PropTypes.number.isRequired,
  searchType: PropTypes.string.isRequired,
  match: PropTypes.shape({
    url: PropTypes.string,
  }).isRequired,
};

export default Status;
