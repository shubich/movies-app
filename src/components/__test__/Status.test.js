import React from 'react';
import { Status } from '../Status';
import { shallow } from 'enzyme';

let props;

describe('Status', () => {
    beforeEach(() => {
        props = {
            match: {
                url: '/film/680',
                params: {
                    id: '680'
                }
            },
            count: '20',
            sort: 'rating',
            handleSort: jest.fn            
        };
    });

    it('renders correctly with props', () => {
        const wrapper = shallow(
            <Status {...props} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with another sort', () => {
        const wrapper = shallow(
            <Status {...props} sort='date' />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly without props', () => {
        const wrapper = shallow(
            <Status/>
        );
        expect(wrapper).toMatchSnapshot();
    });
})