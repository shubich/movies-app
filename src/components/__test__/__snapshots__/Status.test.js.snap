// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status renders correctly with another sort 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Status
    count="20"
    handleSort={[Function]}
    match={
      Object {
        "params": Object {
          "id": "680",
        },
        "url": "/film/680",
      }
    }
    sort="date"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="count"
        >
          20
           movies found
        </div>
        <div
          className="sort"
        >
          <span
            className="sortBy"
          >
            Sort by
          </span>
          <a
            className="link link-primary"
            onClick={[Function]}
          >
            release date
          </a>
          <a
            className="link link-default"
            onClick={[Function]}
          >
            rating
          </a>
        </div>
      </div>,
      "id": "status",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="count"
          >
            20
             movies found
          </div>,
          <div
            className="sort"
          >
            <span
              className="sortBy"
            >
              Sort by
            </span>
            <a
              className="link link-primary"
              onClick={[Function]}
            >
              release date
            </a>
            <a
              className="link link-default"
              onClick={[Function]}
            >
              rating
            </a>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "20",
              " movies found",
            ],
            "className": "count",
          },
          "ref": null,
          "rendered": Array [
            "20",
            " movies found",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="sortBy"
              >
                Sort by
              </span>,
              <a
                className="link link-primary"
                onClick={[Function]}
              >
                release date
              </a>,
              <a
                className="link link-default"
                onClick={[Function]}
              >
                rating
              </a>,
            ],
            "className": "sort",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Sort by",
                "className": "sortBy",
              },
              "ref": null,
              "rendered": "Sort by",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "release date",
                "className": "link link-primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "release date",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "rating",
                "className": "link link-default",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "rating",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="count"
          >
            20
             movies found
          </div>
          <div
            className="sort"
          >
            <span
              className="sortBy"
            >
              Sort by
            </span>
            <a
              className="link link-primary"
              onClick={[Function]}
            >
              release date
            </a>
            <a
              className="link link-default"
              onClick={[Function]}
            >
              rating
            </a>
          </div>
        </div>,
        "id": "status",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="count"
            >
              20
               movies found
            </div>,
            <div
              className="sort"
            >
              <span
                className="sortBy"
              >
                Sort by
              </span>
              <a
                className="link link-primary"
                onClick={[Function]}
              >
                release date
              </a>
              <a
                className="link link-default"
                onClick={[Function]}
              >
                rating
              </a>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "20",
                " movies found",
              ],
              "className": "count",
            },
            "ref": null,
            "rendered": Array [
              "20",
              " movies found",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="sortBy"
                >
                  Sort by
                </span>,
                <a
                  className="link link-primary"
                  onClick={[Function]}
                >
                  release date
                </a>,
                <a
                  className="link link-default"
                  onClick={[Function]}
                >
                  rating
                </a>,
              ],
              "className": "sort",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Sort by",
                  "className": "sortBy",
                },
                "ref": null,
                "rendered": "Sort by",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "release date",
                  "className": "link link-primary",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "release date",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "rating",
                  "className": "link link-default",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "rating",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Status renders correctly with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Status
    count="20"
    handleSort={[Function]}
    match={
      Object {
        "params": Object {
          "id": "680",
        },
        "url": "/film/680",
      }
    }
    sort="rating"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="count"
        >
          20
           movies found
        </div>
        <div
          className="sort"
        >
          <span
            className="sortBy"
          >
            Sort by
          </span>
          <a
            className="link link-default"
            onClick={[Function]}
          >
            release date
          </a>
          <a
            className="link link-primary"
            onClick={[Function]}
          >
            rating
          </a>
        </div>
      </div>,
      "id": "status",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="count"
          >
            20
             movies found
          </div>,
          <div
            className="sort"
          >
            <span
              className="sortBy"
            >
              Sort by
            </span>
            <a
              className="link link-default"
              onClick={[Function]}
            >
              release date
            </a>
            <a
              className="link link-primary"
              onClick={[Function]}
            >
              rating
            </a>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "20",
              " movies found",
            ],
            "className": "count",
          },
          "ref": null,
          "rendered": Array [
            "20",
            " movies found",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="sortBy"
              >
                Sort by
              </span>,
              <a
                className="link link-default"
                onClick={[Function]}
              >
                release date
              </a>,
              <a
                className="link link-primary"
                onClick={[Function]}
              >
                rating
              </a>,
            ],
            "className": "sort",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Sort by",
                "className": "sortBy",
              },
              "ref": null,
              "rendered": "Sort by",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "release date",
                "className": "link link-default",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "release date",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "rating",
                "className": "link link-primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "rating",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="count"
          >
            20
             movies found
          </div>
          <div
            className="sort"
          >
            <span
              className="sortBy"
            >
              Sort by
            </span>
            <a
              className="link link-default"
              onClick={[Function]}
            >
              release date
            </a>
            <a
              className="link link-primary"
              onClick={[Function]}
            >
              rating
            </a>
          </div>
        </div>,
        "id": "status",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="count"
            >
              20
               movies found
            </div>,
            <div
              className="sort"
            >
              <span
                className="sortBy"
              >
                Sort by
              </span>
              <a
                className="link link-default"
                onClick={[Function]}
              >
                release date
              </a>
              <a
                className="link link-primary"
                onClick={[Function]}
              >
                rating
              </a>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "20",
                " movies found",
              ],
              "className": "count",
            },
            "ref": null,
            "rendered": Array [
              "20",
              " movies found",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="sortBy"
                >
                  Sort by
                </span>,
                <a
                  className="link link-default"
                  onClick={[Function]}
                >
                  release date
                </a>,
                <a
                  className="link link-primary"
                  onClick={[Function]}
                >
                  rating
                </a>,
              ],
              "className": "sort",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Sort by",
                  "className": "sortBy",
                },
                "ref": null,
                "rendered": "Sort by",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "release date",
                  "className": "link link-default",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "release date",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "rating",
                  "className": "link link-primary",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "rating",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Status renders correctly without props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Status />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": null,
      "id": "status",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "id": "status",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
